geoh5py/objects/object_base.py

```diff
@@ -211,13 +213,39 @@ def comments(self):
     def default_type_uid(cls) -> uuid.UUID:
         ...
 
+    def clip_by_extent(
+        self, bounds: np.ndarray, parent=None, copy_children: bool = True
+    ) -> ObjectBase | None:
+        """
+        Find indices of vertices within a rectangular bounds.
```

Comment:
**suggestion (maintainability):** make it clear it must be re-implemented
* either add to the doc that default implementation does not clip and returns a copy of the object as-is
* or annotate the method as abstract here, then sub-classes can throw a non-implemented exception

Reply:
Sounds good. I might turn it into abstract actually, and just implement the copy for anything that is not vertex based?

Reply:
Sounds good. I might turn it into abstract actually, and just implement the copy for anything that is not vertex based?

Reply:
Added a warning for all non-vertex entities for now.